### PR TITLE
Addressing issues with certain tasks in task list experiment leading to blank page

### LINKED ISSUE #32714: [WC 6.5 beta1] WCPay task leads to a blank page in the two variants
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The WooCommerce Payments task is not working well with the two variants:
- Clicking the WCPayments task redirects the user to a blank page
- We're not showing the legal message

### Expected behavior

- Clicking in the WCPayments task should redirect the user to the WCPayments onboarding
- We need to display the legal message:
_By setting up, you are agreeing to the [Terms of Service](https://wordpress.com/tos/)_

### Actual behavior

- Clicking the WCPayments task redirects the user to a blank page

### Steps to reproduce

1. Install WC 6.5 Beta 1 and WooCommerce Admin Test Helper 0.7.2
2. Enable usage tracking
3. Install WooCommerce Payments in the Business features step
4. Visit Tools > WCA Test Helper > Experiments and enable woocommerce_tasklist_setup_experiment_2_2022_04
5. Visit the home screen and see that the legal message is missing in the WCPaymens task, click and see the blank page
6. Visit Tools > WCA Test Helper > Experiments and enable woocommerce_tasklist_setup_experiment_1_2022_04
7. Visit the home screen and see that the legal message is missing in the WCPaymens task, click and see the blank page

### WordPress Environment

WooCommerce 6.5 Beta 1

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

It was noticed that certain task actions on the new experiment task lists were not working correctly, primarily the "Get Paid with WooCommerce Payments" task that is added when you install WCPay during the onboarding wizard.

This turned out to require a more significant refactor that I'd originally thought since the new experiment task lists didn't support the `<WooOnboardingTaskListItem />` slot-fill, triggering this issue, and it made sense to refactor the item logic into a separate file as we're doing on the original task list.

Closes #32714  .

### How to test the changes in this Pull Request:

1. Checkout branch on fresh site & install the WooCommerce Admin Test Helper
2. Enable usage tracking
3. Complete OBW and install WooCommerce Payments in the Business features step
4. Visit Tools > WCA Test Helper > Experiments and enable `woocommerce_tasklist_setup_experiment_2_2022_04`
5. Visit the home screen and click "Get paid with WCPay" task to ensure that it redirects as it should.
6. Visit Tools > WCA Test Helper > Experiments and enable woocommerce_tasklist_setup_experiment_1_2022_04
7. Visit the home screen and click "Get paid with WCPay" task to ensure that it redirects as it should.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/two-column-tasks/task-list.tsx [hunk 1]
```diff
@@ -299,30 +272,14 @@ export const TaskList: React.FC< TaskListProps > = ( {
...
-								'woocommerce-task-list__item index-' + index,
-								{
-									'is-complete': task.isComplete,
-									'is-active': task.id === activeTaskId,
-								}
-							);
+
 							return (
-								<TaskItem
+								<TaskListItem
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/two-column-tasks/task-list.tsx:277]: Making use of `TaskListItem` here actually breaks this task list design, it's suppose to look like this:
Screen Shot 2022-04-22 at 12 17 23 PM
But now looks like:
Screen Shot 2022-04-22 at 12 18 39 PM

File: plugins/woocommerce-admin/client/two-column-tasks/task-list-item.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
+/**
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/two-column-tasks/task-list-item.tsx:1]: Thanks for creating a separate component for this, this is indeed much nicer and support the SlotFills.
I end up copying this file pretty well fully in a slightly simplified PR: #32742

File: plugins/woocommerce-admin/client/two-column-tasks/task-list.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+.woocommerce-task-list__item {
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/two-column-tasks/task-list.scss:1]: Will this be needed? These new task list designs don't make use of the expanding/collapsing animations that this CSS is for I believe.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Overall nice work on this @joelclimbsthings, and thanks for getting to the bottom of the issue.
I left several comments, as it does break the design of the first experiment, and also seems to change the design of the second experiment a bit (adding numbers in the task icons).

I did end up creating a slightly simplified fix here: #32742, making use of most of your task-list-item code.

**Comment by joelclimbsthings** [comment]
Thanks @louwie17 ! I definitely should have compared the before/after designs for those experiments, but felt on a bit of a time crunch towards the end. 

Thanks for creating a smaller PR. Just on quick glance it looks like it doesn't utilize the `task-list-item` file for both experiments (for now obvious reasons), but I wonder if we could do so and address the UI issues relatively easily. I was definitely getting a bit confused with all of the redundant code, but I suppose that's expected for the experiments as well. 

Totally fine either way 👍🏻

**Comment by louwie17** [comment]
> Thanks for creating a smaller PR. Just on quick glance it looks like it doesn't utilize the `task-list-item` file for both experiments (for now obvious reasons), but I wonder if we could do so and address the UI issues relatively easily. I was definitely getting a bit confused with all of the redundant code, but I suppose that's expected for the experiments as well.

Totally agree, and I quite like how you went that route. The two designs are a decent amount different so we will have to add several extra options to address for this. Using this PR to address that would be great, I be happy to help with that as well of-course.
Those changes would just be to large to cherry-pick.

**Comment by joelclimbsthings** [comment]
Closing this for now due to conflicts and such, although we may want to borrow part of this for another PR as was mentioned above.

**Pull request closed without merging**